src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -1050,6 +1064,18 @@ private void writeAuthenticationRequired(String realm) {
         write(response);
     }
 
+    private void writeTooManyAuthenticationRequests(String username) {
```

Comment:
pass the request here.

Reply:
what if we wanted to override this implementation and return a different result, status code etc? can this be part of the `RateLimiter`'s responsibility?

Reply:
https://github.com/verygoodsecurity/LittleProxy/blob/vgs-edition/src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java#L1033-L1051

Just was trying to introduce as little changes as possible. Passing HttpRequest will involve code refactoring.